code/iaas/logic/src/main/java/io/cattle/platform/process/storagepool/StoragePoolPurge.java

```diff
@@ -0,0 +1,36 @@
+package io.cattle.platform.process.storagepool;
+
+import io.cattle.platform.core.constants.CredentialConstants;
+import io.cattle.platform.core.constants.StoragePoolConstants;
+import io.cattle.platform.core.model.Credential;
+import io.cattle.platform.core.model.StoragePool;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.process.base.AbstractDefaultProcessHandler;
+
+import javax.inject.Named;
+
+@Named
+public class StoragePoolPurge extends AbstractDefaultProcessHandler {
+
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        StoragePool registry = (StoragePool) state.getResource();
+        if (registry.getKind().equalsIgnoreCase(StoragePoolConstants.KIND_REGISTRY)){
+
+            for (Credential cred : getObjectManager().children(registry, Credential.class)) {
+                if (cred.getRemoved() != null
+                        || !cred.getKind().equalsIgnoreCase(CredentialConstants.KIND_REGISTRY_CREDENTIAL)) {
```

Comment:
Do we need this line? Can a registry have a child credential that is not a registry credential. And if it can, wouldnt we still want to clean it upom registry delete?